### DISCUSSION TITLE
make dirvish-side look like neotree

### DISCUSSION THREAD

**Discussion by thanhvg** [alexluigit/dirvish#100] [Q&A] [answered]
Hi, thanks for a great package. 

Is this possible to remove the file attributes and file size columns in dirvish-side? There must be a setting to make it look like a file tree like neotree or tree-macs isn't it?

**Comment by alexluigit** [comment, marked as answer]
> Is this possible to remove the file attributes and file size columns in dirvish-side?

Of course, you can customize it with `dirvish-side-attributes`

There are other options as well such as `dirvish-side-follow-project-switch`, `dirvish-side-mode-line-format` and all.   `dirvish-side` has only 200+ lines of code, so reading the source code to check out if there are some options might be useful to you should be very easy, I'll improve the doc for this extension in the future though. BTW, feel free to send over a PR if you or anyone want to help with the documentation. 

>  to make it look like a file tree like neotree or tree-macs

There are no specific settings for this.  `dirvish-side` is one of the extension of dirvish, meaning it can work independently or in tandem with other extensions that dirvish provides.  So unlike neotree or treemacs, which requires you to learn a new set of commands and keybindings, you can use it the same way you use dirvish itself. Want a tree-style file explorer? Just try `dirvish-subtree-toggle` on a directory, it works on every dirvish session including a session created by `dirvish-side` command.

**Reply by thanhvg** [reply]
Thanks, to remove file attribute and file size columns `dired-hide-details-mode` must be turned on (https://stackoverflow.com/questions/4115465/emacs-dired-too-much-information). Is there a specific dirvish-side hook or I have to `advice` dirvish-side with `dired-hide-details-mode`?

**Reply by thanhvg** [reply]
one more question with `dirvish-subtree-toggle` when I'm under a subtree, how can I move up to its direct parent, because `dired-up-directory` would just open a buffer of its grand parent directory instead.

**Reply by alexluigit** [reply]
> Is there a specific dirvish-side hook or I have to advice dirvish-side with dired-hide-details-mode?

Aren't those columns (that are displayed before the filename) are hidden by default? It would only happen if you have `dirvish-hide-details` set to nil manually.

> how can I move up to its direct parent

It has not been pushed for now, you can try it out.

```emacs-lisp
(defun dirvish-subtree-beginning ()
  "Jump to beginning of subtree."
  (interactive)
  (when-let ((ov (dirvish-subtree--parent)))
    (goto-char (overlay-start ov))
    (dired-previous-line 1)))
```
